src/sail_json_backend/json.ml

```diff
@@ -98,6 +99,41 @@ let dequote qs =
 
 let string_of_arg = function E_aux (E_id id, _) -> "\"" ^ string_of_id id ^ "\"" | exp -> "exp " ^ string_of_exp exp
 
+(* Function to parse the number at the end of a register name *)
+let parse_reg_name name =
+  try
+    let i = String.rindex name '_' + 1 in
+    (* Find the last underscore, start reading the number just after it *)
+    int_of_string (String.sub name i (String.length name - i))
+  with _ -> -1 (* Return -1 if no number is found or the substring operation fails *)
+
+(* Function to parse the register type from the register name *)
+let json_of_registers () =
+  let regs = Hashtbl.fold (fun name regtype accum -> (name, regtype) :: accum) registers [] in
+  let sorted_regs =
+    List.sort
+      (fun (n1, t1) (n2, t2) ->
+        match String.compare t1 t2 with
+        | 0 ->
+            (* If types are the same, sort by parsed number *)
+            let num1 = parse_reg_name n1 in
+            let num2 = parse_reg_name n2 in
+            if num1 = num2 then String.compare n1 n2 else compare num1 num2
+        | c -> c
+      )
+      regs
+  in
+  let json_regs =
+    List.map
+      (fun (name, regtype) ->
+        "\n  {\n" ^ "    \"name\": \"" ^ name ^ "\",\n" ^ "    \"type\": \"" ^ regtype ^ "\"\n  },"
+      )
+      sorted_regs
+  in
+  let reg_list = String.concat "" json_regs in
+  let final_reg_list = if reg_list = "" then "" else String.sub reg_list 0 (String.length reg_list - 1) in
+  "[ " ^ final_reg_list ^ "\n]"
```

Comment:
I don't think these are used anywhere.

Reply:
Remains of an unused code I forgot to remove